docs/dev/connectors/cassandra.md

```diff
@@ -78,76 +96,195 @@ Note that that enabling this feature will have an adverse impact on latency.
+
+        .keyBy(0)
+        .timeWindow(Time.seconds(5))
+        .sum(1)
+        ;
+
+CassandraSink.addSink(result)
+        .setQuery("INSERT INTO example.wordcount(word, count) values (?, ?);")
+        .setHost("127.0.0.1")
+        .build();
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+CassandraSink.addSink(input.javaStream)
+    .setQuery("INSERT INTO test.wordcount(word, count) values (?, ?);")
+    .setClusterBuilder(new ClusterBuilder() {
+        @Override
+        def buildCluster(builder: Cluster.Builder): Cluster = {
+            builder.addContactPoint("127.0.0.1").build()
+        }
+    })
+    .build()
 {% endhighlight %}
 </div>
 </div>
 
-The Cassandra sinks support both tuples and POJO's that use DataStax annotations.
-Flink automatically detects which type of input is used.
 
-Example for such a Pojo:
+### Cassandra Sink Example for Streaming POJO Data Type
+An example of streaming a POJO data type and store the same POJO entity back to Cassandra. In addition, this POJO implementation needs to follow [DataStax Java Driver Manual](http://docs.datastax.com/en/developer/java-driver/2.1/manual/object_mapper/creating/) to annotate the class as Cassandra connector internally maps each field of this entity to an associated column of the desginated Table using `com.datastax.driver.mapping.Mapper` class of DataStax Java Driver.
+
+Please note that if the upsert query was set, an `IllegalArgumentException` would be thrown with the following error message `Specifying a query is not allowed when using a Pojo-Stream as input.`
+
+For each CQL defined data type for columns, please refer to [CQL Documentation](https://docs.datastax.com/en/cql/3.1/cql/cql_reference/cql_data_types_c.html)
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
+// get the execution environment
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+// get input data by connecting to the socket
+DataStream<String> text = env.socketTextStream(hostname, port, "\n");
 
-@Table(keyspace= "test", name = "mappersink")
-public class Pojo implements Serializable {
+// parse the data, group it, window it, and aggregate the counts
+DataStream<WordCount> result = text
 
-	private static final long serialVersionUID = 1038054554690916991L;
+        .flatMap(new FlatMapFunction<String, WordCount>() {
+            @Override
+            public void flatMap(String value, Collector<WordCount> out) {
+                for (String word : value.split("\\s")) {
+                    if (!word.isEmpty()) {
+                        //Do not accept empty word, since word is defined as primary key in C* table
+                        out.collect(new WordCount(word, 1L));
+                    }
+                }
+            }
+        })
 
-	@Column(name = "id")
-	private long id;
-	@Column(name = "value")
-	private String value;
+        .keyBy("word")
```

Comment:
why is this example different than the previous one about tuples?

Reply:
Thanks. Do you refer difference as for the string splitting or overloading keyBy methods? I will unify string splitting part. As for the keyBy, perhaps user could get to know that it is valid to use keyBy("field name") corresponding to an accessible field in the Pojo class.

Reply:
ah nevermind, i didn't saw that the Tuple example does a keyBy too.